tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/PackageInfo/PackageInfo.cs

```diff
@@ -0,0 +1,106 @@
+    /// Path of the package relative to <see cref="RepoRoot"/> (no leading directory separators).
+    /// </summary>
+    /// <example><c>sdk/storage/storage-blob</c></example>
+    public required string RelativePath { get; init; }
+
+    /// <summary>
+    /// Canonical package identifier (language specific). For example: <c>@azure/storage-blob</c>, <c>Azure.Storage.Blobs</c>, <c>azure-storage-blob</c>.
+    /// </summary>
+    /// <remarks>
+    /// Resolved by language specific helpers when constructing the instance.
+    /// </remarks>
+    public required string PackageName { get; init; }
+
+    /// <summary>
+    /// Logical Azure service name that the package targets (e.g. <c>storage</c>, <c>keyvault</c>, <c>cosmos</c>).
+    /// </summary>
+    public required string ServiceName { get; init; }
+
+    /// <summary>
+    /// Language moniker (e.g. <c>typescript</c>, <c>dotnet</c>, <c>python</c>, <c>java</c>, <c>go</c> ).
+    /// </summary>
+    /// <remarks>
+    /// Used for selecting language specific strategies (sample folder layout, file extensions, version extraction, etc.).
+    /// </remarks>
+    public required string Language { get; init; }
+
+    // Internal strategy delegates filled by language-specific helper at construction time.
+    /// <summary>
+    /// Delegate that returns the absolute path to the directory containing runnable samples for the package.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <c>&lt;PackagePath&gt;/samples</c>. Languages with different conventions (e.g. <c>samples/v12</c>, <c>sample</c>, or nested per-feature
+    /// folders) can override this when creating the <see cref="PackageInfo"/> instance.
+    /// </remarks>
+    internal Func<PackageInfo, CancellationToken, Task<string>> SamplesDirectoryProvider { get; init; } = (pi, ct) => Task.FromResult(Path.Combine(pi.PackagePath, "samples"));
+
+    /// <summary>
+    /// Delegate that resolves the primary source file extension used by generated samples for this package / language.
+    /// </summary>
+    /// <remarks>
+    /// The default value is <c>.txt</c> as a placeholder; language helpers must override to provide a meaningful extension (e.g. <c>.ts</c>, <c>.py</c>, <c>.cs</c>).
+    /// </remarks>
+    internal Func<PackageInfo, string> FileExtensionProvider { get; init; } = pi => ".txt"; // default placeholder
+
+    /// <summary>
+    /// Delegate that returns the current package version string, or <c>null</c> if it cannot be determined cheaply.
+    /// </summary>
+    /// <remarks>
+    /// Version extraction may rely on language specific manifest files (e.g. <c>package.json</c>, <c>AssemblyInfo.cs</c>, <c>setup.cfg</c>, etc.) so this indirection
+    /// allows callers to request a version only when they need it and lets language helpers customize the lookup.
+    /// </remarks>
+    internal Func<PackageInfo, CancellationToken, Task<string?>> VersionProvider { get; init; } = (pi, ct) => Task.FromResult<string?>(null);
```

Comment:
Okay, this I sort of understand could be a lambda, but is reading a package version from a file really that expensive? I feel like it would be fine to just have the factory function get this.

Alternatively, since samples generation does not need the version property as per the [doc comment below](https://github.com/Azure/azure-sdk-tools/pull/12419/files#diff-7866e13c9468250ef501862ee31476b83056bacbf24a9f5da0d8e08798c50da0R103), maybe we could just drop this property for now and revisit if it becomes necessary in future.